src/main/java/com/conveyal/datatools/manager/jobs/MonitorServerStatusJob.java

```diff
@@ -0,0 +1,145 @@
+        this.deployment = deployment;
+        this.instance = instance;
+        this.otpServer = otpServer;
+        status.message = "Checking server status...";
+    }
+
+    @Override
+    public void jobLogic() {
+        long startTime = System.currentTimeMillis();
+        // Get OTP URL for instance to check for availability.
+        String otpUrl = String.format("http://%s/otp", instance.getPublicIpAddress());
+        boolean otpIsRunning = false, routerIsAvailable = false;
+        // Progressively check status of OTP server
+        if (deployment.buildGraphOnly) {
+            // FIXME No need to check that OTP is running. Just check to see that the graph is built.
+//            FeedStore.s3Client.doesObjectExist(otpServer.s3Bucket, );
+        }
+        // First, check that OTP has started up.
+        status.update("Instance status is OK. Waiting for OTP...", 20);
+        while (!otpIsRunning) {
+            LOG.info("Checking that OTP is running on server at {}.", otpUrl);
+            // If the request is successful, the OTP instance has started.
+            otpIsRunning = checkForSuccessfulRequest(otpUrl, 5);
+            if (System.currentTimeMillis() - startTime > TIMEOUT_MILLIS) {
+                status.fail(String.format("Job timed out while monitoring setup for server %s", instance.getInstanceId()));
+                return;
+            }
+        }
+        String otpMessage = String.format("OTP is running on server %s (%s). Building graph...", instance.getInstanceId(), otpUrl);
+        LOG.info(otpMessage);
+        status.update(otpMessage, 30);
+        // Once this is confirmed, check for the existence of the router, which will indicate that the graph build is
+        // complete.
+        String routerUrl = String.format("%s/routers/default", otpUrl);
+        while (!routerIsAvailable) {
+            LOG.info("Checking that router is available (i.e., graph build or read is finished) at {}", routerUrl);
+            // If the request was successful, the graph build is complete!
+            // TODO: Substitute in specific router ID? Or just default to... default.
+            routerIsAvailable = checkForSuccessfulRequest(routerUrl, 5);
+        }
+        status.update(String.format("Graph build completed on server %s (%s).", instance.getInstanceId(), routerUrl), 90);
+        if (otpServer.ec2Info.targetGroupArn != null) {
+            // After the router is available, the EC2 instance can be registered with the load balancer.
+            // REGISTER INSTANCE WITH LOAD BALANCER
+            AmazonElasticLoadBalancing elbClient = AmazonElasticLoadBalancingClient.builder().build();
+            RegisterTargetsRequest registerTargetsRequest = new RegisterTargetsRequest()
+                    .withTargetGroupArn(otpServer.ec2Info.targetGroupArn)
+                    .withTargets(new TargetDescription().withId(instance.getInstanceId()));
+            elbClient.registerTargets(registerTargetsRequest);
```

Comment:
Could this throw an exception if it fails? If so, we want to be sure to catch that and fail the job accordingly.

Reply:
Confirmed that this line can throw an exception. I got this error:

```
com.amazonaws.services.elasticloadbalancingv2.model.InvalidTargetException: The following targets are not in a running state and cannot be registered: 'i-0d76a43db722632ca' (Service: AmazonElasticLoadBalancing; Status Code: 400; Error Code: InvalidTarget; Request ID: 9d7acd6c-1639-4947-8a10-d8c1837d8959)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.handleErrorResponse(AmazonHttpClient.java:1712)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeOneRequest(AmazonHttpClient.java:1367)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeHelper(AmazonHttpClient.java:1113)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.doExecute(AmazonHttpClient.java:770)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.executeWithTimer(AmazonHttpClient.java:744)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.execute(AmazonHttpClient.java:726)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutor.access$500(AmazonHttpClient.java:686)
	at com.amazonaws.http.AmazonHttpClient$RequestExecutionBuilderImpl.execute(AmazonHttpClient.java:668)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:532)
	at com.amazonaws.http.AmazonHttpClient.execute(AmazonHttpClient.java:512)
	at com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient.doInvoke(AmazonElasticLoadBalancingClient.java:2765)
	at com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient.invoke(AmazonElasticLoadBalancingClient.java:2732)
	at com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient.invoke(AmazonElasticLoadBalancingClient.java:2721)
	at com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient.executeRegisterTargets(AmazonElasticLoadBalancingClient.java:2303)
	at com.amazonaws.services.elasticloadbalancingv2.AmazonElasticLoadBalancingClient.registerTargets(AmazonElasticLoadBalancingClient.java:2274)
	at com.conveyal.datatools.manager.jobs.MonitorServerStatusJob.jobLogic(MonitorServerStatusJob.java:147)
```

The job was failed and the DeployJob terminated the instance automatically, so there wasn't even a chance to try to debug what happened. The user logs were uploaded, but don't give any clue as to why the instance wasn't in a running state. Very strange.